### PR TITLE
feat: display same source and destination token warning on widget

### PR DESCRIPTION
**Pull request by mikasackermn**
# Summary

If the source and destination tokens are selected as the same, we should show a warning to the user.

Fixes # (issue)

# How did you test this change?

Select the same token for both the source and destination, and also enter an amount

# Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

File: widget/embedded/src/components/NoResult/NoResult.tsx [hunk 1]
```diff
@@ -38,6 +38,8 @@ export function NoResult(props: PropTypes) {
     fetch
   );
 
+  console.log(info);
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/components/NoResult/NoResult.tsx:41]: Please remove logs

File: widget/embedded/src/hooks/useSwapInput.ts [hunk 2]
```diff
@@ -107,7 +105,14 @@ export function useSwapInput({
       affiliatePercent,
       affiliateWallets,
     } = params;
+    if (areTokensEqual(fromToken, toToken)) {
+      console.log('....');
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/hooks/useSwapInput.ts:109]: Please remove logs

**Comment by Ikari-Shinji-re** [comment]
In the current implementation, the warning appears after the debounce period for the input. If the user changes the input, the warning will be cleared and then reappear after the debounce delay. I suggest leaving the existing logic for the quote unchanged and simply adding a check and warning for this specific state in the relevant component, so the warning displays immediately without delay.

File: widget/embedded/src/components/SameTokensWarning/SameTokensWarning.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import { useQuoteStore } from '../../store/quote';
+import { areTokensEqual } from '../../utils/wallets';
+
+import { Container } from './SameTokensWarning.styles';
+
+export function SameTokensWarning() {
+  const { fromToken, toToken } = useQuoteStore();
+  const showWarningMessage =
+    fromToken && toToken && areTokensEqual(fromToken, toToken);
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/components/SameTokensWarning/SameTokensWarning.tsx:13]: The type of `showWarningMessage` is inferred as `boolean | null`. Using `!!fromToken` and `!!toToken` would ensure the inferred type is strictly `boolean`, making it better aligned with the variable's name.

**Comment by Ikari-Shinji-re** [comment]
Please review the UI design again. The spacing around the newly added component does not align with the design specifications

**Review by Ikari-Shinji-re — APPROVED** [review]

File: widget/embedded/src/components/SameTokensWarning/SameTokensWarning.tsx (hunk 3 above)
Comment (mikasackermn) [reply widget/embedded/src/components/SameTokensWarning/SameTokensWarning.tsx]: Fixed

**Pull request merged**